lib/AST/RequirementMachine/GenericSignatureQueries.cpp

```diff
@@ -525,23 +523,21 @@ bool RequirementMachine::isValidTypeParameter(Type type) const {
 ConformancePath
 RequirementMachine::getConformancePath(Type type,
                                        ProtocolDecl *protocol) {
-  assert(type->isTypeParameter());
-
   auto mutTerm = Context.getMutableTermForType(type->getCanonicalType(),
                                                /*proto=*/nullptr);
   System.simplify(mutTerm);
   verify(mutTerm);
 
-#ifndef NDEBUG
-  auto *props = Map.lookUpProperties(mutTerm);
-  assert(props &&
-         "Subject type of conformance access path should be known");
-  assert(!props->isConcreteType() &&
-         "Concrete types do not have conformance access paths");
-  auto conformsTo = props->getConformsTo();
-  assert(std::find(conformsTo.begin(), conformsTo.end(), protocol) &&
-         "Subject type of conformance access path must conform to protocol");
-#endif
+  if (CONDITIONAL_ASSERT_enabled()) {
+    auto *props = Map.lookUpProperties(mutTerm);
+    ASSERT(props &&
+           "Subject type of conformance access path should be known");
+    ASSERT(!props->isConcreteType() &&
+           "Concrete types do not have conformance access paths");
+    auto conformsTo = props->getConformsTo();
+    CONDITIONAL_ASSERT(std::find(conformsTo.begin(), conformsTo.end(), protocol) &&
```

Comment:
Not much point to having `CONDITIONAL_ASSERT` inside a `CONDITIONAL_ASSERT_enabled` block.

Reply:
Ah, good catch.